apps/src/blockly/addons/cdoUtils.js

```diff
@@ -6,6 +6,10 @@ import DCDO from '@cdo/apps/dcdo';
 import {APP_HEIGHT} from '@cdo/apps/p5lab/constants';
 import {SOUND_PREFIX} from '@cdo/apps/assetManagement/assetPrefix';
 
+export function loadBlocksToWorkspace(workspace, xml, stateToLoad) {
+  return Blockly.Xml.domToBlockSpace(workspace, xml);
```

Comment:
Today, we only ever reach this point with `workspace` and `xml`. As we add JSON sources to projects, we'd be able to use that state like `Blockly.serialization.workspaces.load(stateToLoad, workspace)`. We will always want to support cases where we have xml and not json state. In that case, we can either call `domToBlockSpace` like we do above, or consider other paths like converting the xml to json and then using the same `load` method above.

Reply:
Why do we need 3 parameters here? Would there ever be a time when both xml and stateToLoad are both passed and used? Can we use 2 parameters like (workspace, state) and just check to see if state is json or xml?

Reply:
We need 3 parameters because we're planning to save the json in parallel with the xml, at least for a time. This way, if we run into any issues loading `stateToLoad` we'll be able to fall back on the xml.

For context, see the table at this heading: https://docs.google.com/document/d/1V367MAAYU0JXETVX6l48_6HW2h93WDzdljJc1ohLZPc/edit#heading=h.33jxpkjaxa77

One thing that I like about this is that we'd be able to simply check that `stateToLoad` exists in order to be able to expect that it's JSON.

Reply:
Gotcha - that makes sense. Thanks!